lib/Transforms/Scalar/PromoteVerifierCalls.cc

```diff
@@ -171,28 +210,57 @@ bool PromoteVerifierCalls::runOnFunction(Function &F) {
                fn->getName().equals("llvm.invariant") ||
                /** my suggested name for pagai invariants */
                fn->getName().equals("pagai.invariant"))) {
-      Function *nfn;
-      if (fn->getName().equals("__VERIFIER_assume"))
-        nfn = m_assumeFn;
-      else if (fn->getName().equals("llvm.invariant"))
-        nfn = m_assumeFn;
-      else if (fn->getName().equals("pagai.invariant"))
-        nfn = m_assumeFn;
-      else if (fn->getName().equals("__VERIFIER_assert"))
-        nfn = m_assertFn;
-      else if (fn->getName().equals("__VERIFIER_assert_not"))
-        nfn = m_assertNotFn;
-      else if (fn->getName().equals("__CPROVER_assume"))
-        nfn = m_assumeFn;
-      else
-        continue;
+      auto arg0 = CS.getArgument(0);
+
+      CallInst *nfnCi = nullptr, *chkCi = nullptr;
+      if (auto partialFn = extractPartialFnCall(arg0)) {
+        // Selects proper synthesis call.
+        Function *nfn;
+        if (fn->getName().equals("__VERIFIER_assume"))
+          nfn = m_synthAssumeFn;
+        else if (fn->getName().equals("__VERIFIER_assert"))
+          nfn = m_synthAssertFn;
+        else
+          assert(0);
+
+        // Generates code.
+        IRBuilder<> Builder(F.getContext());
+        Builder.SetInsertPoint(&I);
+        auto *sentinel = ConstantInt::get(partialFn->getType(), 1);
```

Comment:
in C 0 means false and non 0 means true. I think you are assuming that 1 means true instead